app/Http/Controllers/MagazineIssueController.php

```diff
@@ -24,13 +25,7 @@ public function latest()
 
     public function show($id, $slug = null, $isRequestForLatest = false)
     {
-        $issues = $this->repository->published()->ordered()->get();
-        $item = $issues->where('id', (int) $id)->first();
-
-        if (!$item) {
-            abort(404);
-        }
-
+        $item = $this->repository->findOrFail($id);
```

Comment:
This line is equivalent to getting a model by id or throwing a 404 if it doesn't exist.